packages/editor-ui/src/components/TextWithHighlights.vue

```diff
@@ -0,0 +1,39 @@
+<script lang="ts" setup>
+import type { PropType } from 'vue';
+import { GenericValue } from 'n8n-workflow';
+
+const props = defineProps({
+	content: {
+		type: [Object, String, Number] as PropType<GenericValue>,
+	},
+	search: {
+		type: String,
+	},
+});
+
+const splitTextBySearch = (text = '', search = ''): { tag: 'span' | 'mark'; content: string }[] => {
+	if (!search) {
+		return [
+			{
+				tag: 'span',
+				content: text,
+			},
+		];
+	}
+
+	const pattern = new RegExp(`(${search})`, 'g');
```

Comment:
I think we can get away with only one Regex here:
```
const pattern = new RegExp(`(${search})`, 'gi');
const splitText = text.split(pattern);
```

Reply:
oh ya that's  a mistake